source/constructs/api/template/service.py

```diff
@@ -167,10 +171,231 @@
     if identifier.type == IdentifierType.CUSTOM.value and identifier.header_keywords and '""' in identifier.header_keywords[1:-1].split(","):
         raise BizException(MessageEnum.TEMPLATE_HEADER_KEYWORDS_EMPTY.get_code(),
                            MessageEnum.TEMPLATE_HEADER_KEYWORDS_EMPTY.get_msg())
+
+
+def export_identify(key):
+    default_sheet = "Sheet"
+    workbook = openpyxl.Workbook()
+    sheet = workbook.create_sheet(const.BATCH_SHEET, index=0)
+    if default_sheet in workbook.sheetnames and len(workbook.sheetnames) > 1:
+        origin_sheet = workbook[default_sheet]
+        workbook.remove(origin_sheet)
+    sheet.append(const.EXPORT_IDENTIFY_HEADER)
+    result = crud.get_all_identifiers()
+    props_list = crud.get_identifier_prop_mapping()
+    props_mapping = __convert_prop_list_2_mapping(props_list)
+    for row_num, row_data in enumerate(result, start=2):
+        category = props_mapping.get(f"{row_data[0]}-1")
+        label = props_mapping.get(f"{row_data[0]}-2")
+        # del row_data[0]
+        for col_num, cell_value in enumerate(row_data, start=1):
+            if col_num == 1:
+                continue
+            if col_num == 5 or col_num == 6:
+                sheet.cell(row=row_num, column=col_num - 1).value = cell_value.replace('\"', '')[1:-1] if cell_value else cell_value
+            else:
+                sheet.cell(row=row_num, column=col_num - 1).value = cell_value
+            if category:
+                sheet.cell(row=row_num, column=8, value=category)
+            if label:
+                sheet.cell(row=row_num, column=9, value=label)
+    workbook.active = 0
+    file_name = f"identify_{key}.xlsx"
+    tmp_file = f"{tempfile.gettempdir()}/{file_name}"
+    report_file = f"{const.IDENTIFY_REPORT}/{file_name}"
+    workbook.save(tmp_file)
+    stats = os.stat(tmp_file)
+    if stats.st_size < 6 * 1024 * 1024:
+        __s3_client.upload_file(tmp_file, admin_bucket_name, report_file)
+    else:
+        concurrent_upload2s3(admin_bucket_name, report_file, tmp_file, __s3_client)
+    os.remove(tmp_file)
+    method_parameters = {'Bucket': admin_bucket_name, 'Key': report_file}
+    pre_url = __s3_client.generate_presigned_url(
+        ClientMethod="get_object",
+        Params=method_parameters,
+        ExpiresIn=60
+    )
+    return pre_url
+
+def __convert_prop_list_2_mapping(list):
+    mapping = {}
+    for item in list:
+        mapping[f"{item[0]}-{item[1]}"] = item[2]
+    return mapping
+
+def delete_report(key):
+    __s3_client.delete_object(Bucket=admin_bucket_name, Key=f"{const.BATCH_CREATE_IDENTIFIER_REPORT_PATH}/{key}.xlsx")
+
+
+def query_batch_status(filename: str):
+    success, warning, failed = 0, 0, 0
+    file_key = f"{const.BATCH_CREATE_IDENTIFIER_REPORT_PATH}/{filename}.xlsx"
+    response = __s3_client.list_objects_v2(Bucket=admin_bucket_name, Prefix=const.BATCH_CREATE_IDENTIFIER_REPORT_PATH)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The API method `list_objects_v2` returns paginated results instead of all results. Consider using the pagination API or checking one of the following keys in the response to verify that all results were returned: `ContinuationToken`, `IsTruncated`, `NextContinuationToken`, `NextToken`.

[Learn more](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/paginators.html)